src/main/java/gregtech/api/unification/ore/OrePrefix.java

```diff
@@ -581,12 +581,51 @@ public String getAlternativeOreName() {
     public String getLocalNameForItem(Material material) {
         String specifiedUnlocalized = "item." + material.toString() + "." + this.name;
         if (LocalizationUtils.hasKey(specifiedUnlocalized)) return LocalizationUtils.format(specifiedUnlocalized);
-        String unlocalized = "item.material.oreprefix." + this.name;
+        String unlocalized = findUnlocalizedName(material);
         String matLocalized = material.getLocalizedName();
         String formatted = LocalizationUtils.format(unlocalized, matLocalized);
         return formatted.equals(unlocalized) ? matLocalized : formatted;
     }
 
+    private String findUnlocalizedName(Material material) {
+
+        String name = "";
+
+        // Avoid NPE from Materials.NULL
+        if(!material.getUnlocalizedName().equals("material.null") && material.hasProperty(PropertyKey.POLYMER)) {
+            if(this.equals(plate)) {
```

Comment:
Better implementation would be via Map or similar where you could jus call getOrDefault()

Reply:
Yeah this is awful. Why not just
```java
private String findUnlocalizedName(Material material) {
    if (material.getUnlocalizedName().equals("material.null")) {
        return "Epic localization fail";
    }
    if (material.hasProperty(PropertyKey.POLYMER)) {
        return String.format("item.material.oreprefix.polymer.%s", this.name);
    }
    return String.format("item.material.oreprefix.%s", this.name);
}
```
And to make it even better you you make a `ILocalizedMaterialPropertyKey`. Which would have a method like `getLangKey(OrePrefix orePrefix, Material material)`.

Reply:
I would be bit careful of returning "Epic localization fail". But if using another level of dot formatting in lang file is OK, then this would be better as oreprefix itself will stay same.